execution_engine/src/shared/storage_costs.rs

```diff
@@ -27,9 +24,15 @@ impl StorageCosts {
     }
 
     /// Calculates gas cost for storing `bytes`.
-    pub fn calculate_gas_cost(&self, bytes: usize) -> Gas {
-        let value = U512::from(self.gas_per_byte) * U512::from(bytes);
-        Gas::new(value)
+    ///
+    /// Returns wrapped [`Gas`] if calculation did not overflow. This method will return [`None`] if
+    /// the result would overflow [`Gas`] range - in this case the contract execution logic should
+    /// exit early with a gas limit error.
+    pub fn calculate_gas_cost(&self, bytes: usize) -> Option<Gas> {
```

Comment:
This looks like duplicate code [from above][1].  Here too, shouldn't we be returning a `Result` rather than an option?

[1]: https://github.com/casper-network/casper-node/pull/1968/files#diff-7201cf01479c2c4eecedce0e5537db3472585761a185dc2aa994dc2258855060R134